examples/baseball.py

```diff
@@ -100,41 +105,32 @@ def partially_pooled(at_bats):
     and $kappa ~ Pareto(1, 1.5)$.
 
     :param (torch.Tensor) at_bats: Number of at bats for each player.
+    :param (torch.Tensor) hits: Number of hits for the given at bats.
     :return: Number of hits predicted by the model.
     """
     num_players = at_bats.shape[0]
     m = pyro.sample("m", Uniform(at_bats.new_tensor(0), at_bats.new_tensor(1)))
     kappa = pyro.sample("kappa", Pareto(at_bats.new_tensor(1), at_bats.new_tensor(1.5)))
     phi_prior = Beta(m * kappa, (1 - m) * kappa).expand_by([num_players]).independent(1)
     phi = pyro.sample("phi", phi_prior)
-    return pyro.sample("obs", Binomial(at_bats, phi))
+    return pyro.sample("obs", Binomial(at_bats, phi), obs=hits)
 
 
-def partially_pooled_with_logit(at_bats):
+def partially_pooled_with_logit(at_bats, hits):
     r"""
     Number of hits has a Binomial distribution with a logit link function.
     The logits $\alpha$ for each player is normally distributed with the
     mean and scale parameters sharing a common prior.
 
     :param (torch.Tensor) at_bats: Number of at bats for each player.
+    :param (torch.Tensor) hits: Number of hits for the given at bats.
     :return: Number of hits predicted by the model.
     """
     num_players = at_bats.shape[0]
     loc = pyro.sample("loc", Normal(at_bats.new_tensor(-1), at_bats.new_tensor(1)))
     scale = pyro.sample("scale", HalfCauchy(scale=at_bats.new_tensor(1)))
     alpha = pyro.sample("alpha", Normal(loc, scale).expand_by([num_players]).independent(1))
-    return pyro.sample("obs", Binomial(at_bats, logits=alpha))
-
-
-def conditioned_model(model, at_bats, hits):
```

Comment:
How does this work when we need to sample from the posterior predictive? Does this rely on us calling the model with `hits=None`. Do the results remain the same with this refactoring?

Reply:
It seems so. Why do you expect that things will be different? I'll check it btw.

Reply:
@neerajprad They give identical results for the fully pooled model.

Reply:
Yes, passing `obs=None` makes it a latent site, not observe site. The difference between two versions is at the `_RETURN` node. With conditioning, model returns sample at `obs`. With `obs=None`, model returns `None`. I remember so.

Reply:
I think that's fine for as long as we aren't using the return value.